clippy_lints/src/formatting.rs

```diff
@@ -135,10 +134,10 @@ fn check_assign(cx: &EarlyContext<'_>, expr: &ast::Expr) {
 }
 
 /// Implementation of the `SUSPICIOUS_ELSE_FORMATTING` lint for weird `else`.
-fn check_else(cx: &EarlyContext<'_>, expr: &ast::Expr) {
+fn check_else(cx: &EarlyContext<'_>, expr: &Expr) {
     if_chain! {
-        if let Some((then, &Some(ref else_))) = unsugar_if(expr);
-        if is_block(else_) || unsugar_if(else_).is_some();
+        if let ExprKind::If(_, then, Some(else_)) = &expr.node;
+        if is_block(else_) || is_if(else_);
```

Comment:
There are multiple occurences of this function in a `||` or `&&` expression. Removing this function would make the code way more complicated IMO

Reply:
ah